2015/proposals/banaszkiewicz-piotr.md

```diff
@@ -0,0 +1,158 @@
+# Installation-test scripts: aggregation server, script enhancements
+
+## Abstract
+
+While Software Carpentry workshops gather a lot of feedback regarding
+instructors' teaching, not much attention was drawn to students' computers and
+their issues with installation of open source software used during workshops.
+
+This project will focus on bringing a working server for aggregating of
+Software Carpentry installation testing scripts with additional goal of
+providing easier and better experience for students using the script itself.
+
+## Features
+
+The server should be able to aggregate incoming (via REST API) data from
+installation testing script.
+
+The server should allow Software Carpentry admins to view statistics from
+uploaded data.
+
+Installation script should upload diagnostic data consisting of:
+
+* operating system information
+* installed packages and their versions
+* failed packages checks and error messanges
+
+Additionally, the data may be matched against workshop
+[reference package list](https://github.com/wking/swc-setup-installation-test/issues/2).
+
+## Technical Details
+
+The server will be implemented using [Django](https://www.djangoproject.com/)
+and [SQLite](https://docs.djangoproject.com/en/1.7/ref/databases/) database
+for lower footprint and easier maintainance (and because I feel comfortable
+using them).
+
+API, in a RESTful fashion, will be implemented using
+[django-tastypie](http://tastypieapi.org/).
+
+Rough database structure:
+
+* system information table
+    * operating system family (Windows / MacOSX / Linux / Other)
+    * OS version (8.1 / 10.9 / Ubuntu / ?)
+    * Hardware architecture (x86_64 / x86 / Other)
+* package checks table
+    * package name
+    * requested package version
+    * found package version
+    * failed check?
+    * fail reason
+
+Every entry in both tables will be additionally assigned a
+[universally unique identifier](http://en.wikipedia.org/wiki/Universally_unique_identifier).
+Every single UUID will correspond to one diagnostic data upload.
```

Comment:
On Thu, Mar 19, 2015 at 02:07:01PM -0700, Piotr Banaszkiewicz wrote:
> Now I think that it could be "installation test entry" containing
> OS-related information and maybe submission-related, too (IP?
> Workshop slug?).

This sounds reasonable to be, but…

> This way I could use that entry's ID as a foreign key in the other
> table ("package checks").

I don't understand why the fields that are or aren't in the
os-information table impacts whether or not you can use a foreign key.
If you have related tables, you should be using foreign keys to link
them together.  If you don't have a good reason to use a UUID for the
target of the foreign key, you should stick to auto-incremented
integers.  Again, I'm not saying that there *aren't* good reasons to
use UUID targets, just that I haven't heard you explain one yet.